packages/components/src/components/segmented-control/segmented-control.tsx

```diff
@@ -83,9 +86,10 @@ export class SegmentedControl {
 
     render() {
         return (
-            <div aria-value={this.selectedValue} aria-label='segmented control' class='group'>
+            <div aria-value={this.selectedValue} aria-label='segmented control' class={`group ${this.error ? 'error' : ''} ${this.required ? 'required' : ''}`}>
```

Comment:
Why are you conditionally setting the error and required classes on the group element?